src/bg_mpl_stylesheets/colors.py

```diff
@@ -0,0 +1,28 @@
+from enum import Enum
+
+
+class Colors(str, Enum):
+    bg_blue = "#0B3C5D"
+    bg_red = "#B82601"
+    bg_green = "#1C6B0A"
+    bg_light_blue = "#328CC1"
+    bg_light_grey = "#A8B6C1"
+    bg_yellow = "#D9B310"
+    bg_brown = "#6C5050"
+    bg_burgundy = "#76323F"
+    bg_olive_green = "#626E60"
+    bg_muted_olive = "#918770"
+    bg_beige = "#C09F80"
+    bg_grey = "#B0B0B0FF"
+    # Add more colors as needed
+
+    @classmethod
+    def get_color_name(cls, hex_value: str) -> str:
```

Comment:
Do we actually need these.  It seems that we are not making the best use of enums.  For example, for usage, wouldn't something like this be cleaner?:
```
 import bg_mpl_stylesheets.colors.Colors as bgcolors

for color in bg.colors:
    ax.plot(x, y + offset * i, label=color.name, color=color.value, linestyle="-")
```
and if we want to address explicitly, something like
```
bgblue = bgcolors.bg_blue
ax.plot(x, y + offset * i, label=bgblue.name, color=bgblue.value, linestyle="-")
```

I see that we would need the hex to name converter when using the cycler as a cycle unless we could figure out how to make the cycler elements the enum values.


```

Reply:
```
class Colors(str, Enum):
    bg_blue = "#0B3C5D"
    bg_red = "#B82601"
    bg_green = "#1C6B0A"
    bg_light_blue = "#328CC1"
    bg_light_grey = "#A8B6C1"
    bg_yellow = "#D9B310"
    bg_brown = "#6C5050"
    bg_burgundy = "#76323F"
    bg_olive_green = "#626E60"
    bg_muted_olive = "#918770"
    bg_beige = "#C09F80"
    bg_grey = "#B0B0B0FF"
```

Right, this would be it based on your suggestion above.

The purpose of `get_color_name` was for testing below and changing the legends from hex to color names:

In the example code under `example/color_cycles.py`, we are testing the hard coded `bg-style`.

<img width="700" alt="Screenshot 2024-08-21 at 4 49 11 PM" src="https://github.com/user-attachments/assets/7086c3f7-eb84-4df3-b50a-373acd880d13">

```python
cycle = plt.rcParams["axes.prop_cycle"].by_key()["color"]

# plot the color cycles and corresponding color codes.
for i, hex in enumerate(cycle):
    color_name = Colors.get_color_name(hex)
    ax.plot(x, y + offset * i, label=color_name, color=hex, linestyle="-")
```

If do not have the `get_color_name` class method, then we do not have access to the color name since the `hex` a string.

If retrieving the color name from a string is not needed (legends for the the example code), then, we can remove `get_color_name`

Reply:
Is there no way to get that figure as it is but without the `get_color_name` method but using the enum properties?   That may be true, but something like replacing the hex hard code with the enum values could fix it?   It is still "immutable" in the way that someone could extend the colors enum and it wouldn't change the cycler (which is what we seek for the immutability).  We could also test it with hard coded hex's so if someone changes the hex of one of the used enums the test would break.